hyp3_gamma/insar/unwrapping_geocoding.py

```diff
@@ -4,18 +4,65 @@
 import logging
 import os
 
+import numpy as np
 from hyp3lib.execute import execute
 from hyp3lib.getParameter import getParameter
+from osgeo import gdal
 
 log = logging.getLogger(__name__)
 
 
+def get_minimum_value_gamma_dtype(gammadtype):
+    """
+    gamma datatype  data type:
+             0: RASTER 8 or 24 bit uncompressed raster image, SUN (*.ras), BMP:(*.bmp), or TIFF: (*.tif)
+             1: SHORT integer (2 bytes/value)
+             2: FLOAT (4 bytes/value)
+             3: SCOMPLEX (short complex, 4 bytes/value)
+             4: FCOMPLEX (float complex, 8 bytes/value)
+             5: BYTE
+
+    """
+    if gammadtype in [0, 5]:
+        dt = np.int8
+        return np.iinfo(dt).min
+    elif gammadtype in [1]:
+        dt = np.int16
+        return np.iinfo(dt).min
+    elif gammadtype in [2, 3]:
+        dt = np.float32
+        return np.finfo(dt).min
+    else:
+        dt = np.float64
+        return np.finfo(dt).min
+
+
+def setnodata(file, nodata):
+    """
+    The output geotiff files produced by gamma package always has 0.0 as nodata value.
+    This function changes the nodata value in the geotiff file.
+
+    """
+    ds = gdal.Open(file, gdal.GA_Update)
+    for i in range(1, ds.RasterCount + 1):
+        band = ds.GetRasterBand(i)
+        band_data = band.ReadAsArray()
+        mask = band.GetMaskBand()
+        mask_data = mask.ReadAsArray()
+        band_data[mask_data == 0] = nodata
+        band.WriteArray(band_data)
+        band.SetNoDataValue(float(nodata))
```

Comment:
```suggestion
    for i in range(1, ds.RasterCount + 1):
        band = ds.GetRasterBand(i)
        band_data = band.ReadAsArray()
        mask_data = band.GetMaskBand().ReadAsArray()
        
        band_data[mask_data == 0] = nodata
        band.WriteArray(band_data)
        band.SetNoDataValue(float(nodata))
```

I'm also unsure why the `float()` around `nodata` is needed -- I'd expect the caller to pass a valid nodata value

Reply:
nodata is type of numpy.float32, SetNoDataValue requires the float type. so We need float() there. When I debug the progam, I remember one time mask_data = band.GetMaskBand().ReadAsArray() does not work, this is why I use two statements instead of one.

Reply:
Ah okay, so does the cast to float work with integer types? 

This pattern
```python
nodata = get_minimum_value_gamma_dtype(type_)
setnodata(outname, nodata)
```
can return int or float types, so I would suspect `setnodata` with `float(dtype)` wouldn't work for integer data types. Or does the conversion to tif from GAMMA format always create float tifs? If so, the `get_minimum_value_gamma_dtype` could just convert everything to `float` before it returns.

Reply:
I tested that float() works for int. I didn't check if GAMMA format always create float tifs.

Reply:
:+1: I think I'd move the `float()` into `get_minimum_value_gamma_dtype` so that `setnodata` doesn't hide that magic

Reply:
Actually scratch that, it's fine -- GDAL apparently typically assumes the nodata value is a `float`. E.g., [`gdal_edit.py`](https://github.com/OSGeo/gdal/blob/master/gdal/swig/python/gdal-utils/osgeo_utils/gdal_edit.py#L135)

Reply:
do you mean we leave float() there?

Reply:
Yeah, leave the float there -- seems to be a GDAL thing. :-D

Reply:
OK.